web/src/engine/main/src/keymanEngine.ts

```diff
@@ -190,6 +194,42 @@ export default class KeymanEngine<ContextManager extends ContextManagerBase, Har
 
     return report;
   }
+
+  private refreshModel() {
+    const kbd = this.contextManager.activeKeyboard;
+    const model = this.modelCache.modelForLanguage(kbd.metadata.langId);
+
+    if(this.processor.activeModel != model) {
+      if(this.processor.activeModel) {
+        this.processor.languageProcessor.unloadModel();
+      }
+
+      if(model) {
+        this.processor.languageProcessor.loadModel(model);
+      }
+    }
+  }
+
+  // API methods
+
+  // 17.0: new!  Only used by apps utilizing app/webview and one app/browser test page.
+  addModel(model: ModelSpec) {
+    this.modelCache.register(model);
+
+    if(model.languages.indexOf(this.contextManager.activeKeyboard.metadata.langId) != -1) {
+      this.refreshModel();
+    }
+  }
+
+  // 17.0: new!  Only used by apps utilizing app/webview and one app/browser test page.
+  removeModel(modelId: string) {
+    this.modelCache.deregister(modelId);
+
+    // Is it the active model?
+    if(this.processor.activeModel && this.processor.activeModel.id == modelId) {
+      this.processor.languageProcessor.unloadModel();
+    }
+  }
```

Comment:
We need good API documentation if these are being published, ideally a javadoc style comment for eacch endpoint as well as the corresponding docs in help.keyman.com

Reply:
I don't think we plan to publish these outright as of yet; after all, they're only really meant for use by our mobile apps.

That said, it is a decent prototype toward something we _could_ publish, so the least I can do is provide that Javadoc-style comment now - that's definitely a fair request.